spinc/cmd/find.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2020, Square, Inc.
+
+package cmd
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/square/spincycle/v2/proto"
+	"github.com/square/spincycle/v2/spinc/app"
+)
+
+const (
+	findLimitDefault = 10 // limit to 10 requests by default
+
+	// formatting for outputing request info
+	findReqColLen   = 40
+	findIdColLen    = 20
+	findUserColLen  = 9
+	findStateColLen = 9
+
+	findTimeFmt    = "YYYY-MM-DD HH:MM:SS UTC" // expected time input format
+	findTimeFmtStr = "2006-01-02 15:04:05 MST" // expected time input format as the actual format string (input to time.Parse)
+)
+
+var (
+	findTimeColLen = len(findTimeFmt)
+	findUtcIndex   = strings.Index(findTimeFmt, "UTC")
+)
+
+type Find struct {
+	ctx app.Context
+
+	local  bool
+	filter proto.RequestFilter
+}
+
+func NewFind(ctx app.Context) *Find {
+	return &Find{
+		ctx: ctx,
+	}
+}
+
+func (c *Find) Prepare() error {
+	/* Parse. */
+	// See command usage for details about each filter
+	validArgs := map[string]bool{
+		"timezone": true,
+
+		"type":   true,
+		"states": true,
+		"user":   true,
+		"since":  true,
+		"until":  true,
+		"limit":  true,
+		"offset": true,
+	}
+	args := map[string]string{}
+	for _, arg := range c.ctx.Command.Args {
+		split := strings.SplitN(arg, "=", 2)
+		if len(split) != 2 {
+			return fmt.Errorf("Invalid command arg: %s: split on = produced %d values, expected 2 (filter=value)", arg, len(split))
+		}
+		arg := split[0]
+		value := split[1]
+
+		if !validArgs[arg] {
+			return fmt.Errorf("Invalid arg '%s'", arg)
+		}
+		if _, ok := args[arg]; ok {
+			return fmt.Errorf("Filter '%s' specified multiple times", arg)
+		}
+		args[arg] = value
+
+		if c.ctx.Options.Debug {
+			app.Debug("arg '%s'='%s'", arg, value)
+		}
+	}
+
+	/* Process some args. */
+	var err error
+
+	local := false
+	switch args["timezone"] {
```

Comment:
the input here should be case insensitive. Like "UTC" and "utc" should both be accepted